enterprise/server-enterprise/src/main/java/org/neo4j/server/enterprise/EnterpriseServerConfigLoader.java

```diff
@@ -36,7 +36,10 @@
     public Config loadConfig( File configFile, File legacyConfigFile, Log log, Pair<String,String>... configOverrides )
     {
         Config config = super.loadConfig( configFile, legacyConfigFile, log, configOverrides );
-        config.registerSettingsClasses( asList( HaSettings.class, ClusterSettings.class ) );
+        if ( config.get( EnterpriseServerSettings.mode ).equals( "HA" ) )
```

Comment:
I was not able to create a testable solution. Although I think you are right in principle, it is just too messy to clean it up and create something reasonable and testable. I will have to leave it as it is.